### PR TITLE
[SPARK-38847][CORE][SQL][SS] Introduce a `viewToSeq` function for `KVUtils` to close `KVStoreIterator` in time

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
There are many codes in spark that convert `KVStoreView` into scala `List`, and these codes will not close `KVStoreIterator`, these resources are mainly recycled by `finalize()` method implemented in `LevelDB` and `RockSB`, this makes `KVStoreIterator` resource recycling unpredictable. 

This pr introduce a `viewToSeq` function for `KVUtils`, this function will convert all data in the `KVStoreView` into scala `List` and close `KVStoreIterator` in time.

### Why are the changes needed?
Add a general function to convert `KVStoreView` into scala `List` and close `KVStoreIterator` in time.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/status/KVUtils.scala [hunk 1]
```diff
@@ -92,6 +92,16 @@ private[spark] object KVUtils extends Logging {
     }
   }
 
+  /** Turns a KVStoreView into a Scala sequence. */
+  def viewToSeq[T](view: KVStoreView[T]): Seq[T] = {
+    val iter = view.closeableIterator()
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/status/KVUtils.scala:97]: Maybe `Utils.tryWithResource`.

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/status/KVUtils.scala]: done

**Review by srowen — APPROVED** [review]
Yep seems like a decent idea.

**Comment by srowen** [comment]
Merged to master

**Comment by LuciferYang** [comment]
thanks @srowen

**Pull request closed without merging**